pilot/pkg/xds/ads_test.go

```diff
@@ -979,3 +984,49 @@ func TestPushQueueLeak(t *testing.T) {
 	ds.Discovery.AdsPushAll(&model.PushRequest{Push: ds.PushContext()})
 	p.Cleanup()
 }
+
+func TestDistribution(t *testing.T) {
+	xds.ResetConnectionNumberForTest()
+	s := xdsfake.NewFakeDiscoveryServer(t, xdsfake.FakeOptions{})
+	expectNonce := func(nonce, ty string) {
+		t.Helper()
+		assert.EventuallyEqual(t, func() string {
+			return s.Discovery.StatusReporter.QueryLastNonce("test.default-1", ty)
+		}, nonce[:xds.VersionLen])
+	}
+
+	ledger := ledger.Make(time.Minute)
+	ledger.Put("key", "value") // If there is no config, ledger would be empty
+	s.Env().SetLedger(ledger)
+	reporter := &distribution.Reporter{
+		UpdateInterval: features.StatusUpdateInterval,
+	}
+	reporter.Init(s.Env().GetLedger(), test.NewStop(t))
+	s.Discovery.StatusReporter = reporter
+
+	ads := s.ConnectADS().WithType(v3.ClusterType)
+	// Subscribe to clusters
+	res1 := ads.RequestResponseAck(t, &discovery.DiscoveryRequest{ResourceNames: []string{"fake-cluster"}})
+	expectNonce(res1.Nonce, v3.ClusterType)
+	// Types we are not subscribed to are updated immediately
+	expectNonce(res1.Nonce, v3.RouteType)
+
+	// Send a push
+	s.Discovery.Push(&model.PushRequest{Full: true, Push: s.Env().PushContext()})
+	res := ads.ExpectResponse(t)
+	// Not yet ACKed, should return last one
+	expectNonce(res1.Nonce, v3.ClusterType)
```

Comment:
I apologize for accidentally point to the wrong line - here is where we are supposed to expect for route `res` (NOT res1)

Reply:
Let me test it

Reply:
Sorry i do think we should expect route nonce after the cluster acked

Reply:
can we have a direct talk on slack.  it is very unefficient on github back and forth almost every day